generator/vulkan/render.zig

```diff
@@ -770,8 +770,13 @@ fn Renderer(comptime WriterType: type) type {
 
                 try self.writer.writeAll(" = .");
                 try self.writeIdentifierWithCase(.snake, stype["VK_STRUCTURE_TYPE_".len..]);
-            } else if (field.field_type == .name and !container.is_union and mem.eql(u8, "VkBool32", field.field_type.name) and isFeatureStruct(name, container.extends)) {
-                try self.writer.writeAll(" = FALSE");
+            } else if (field.field_type == .name) {
+                const field_type_name = field.field_type.name;
+                if (!container.is_union and mem.eql(u8, "VkBool32", field_type_name) and isFeatureStruct(name, container.extends)) {
+                    try self.writer.writeAll(" = FALSE");
+                } else if (field.is_optional and self.isFlags(field_type_name)) {
+                    try self.writer.writeAll(" = .{}");
+                }
```

Comment:
What about the other field types like pointers and enums? I think if a pointer is optional we can initialize it will `null` just fine.

Reply:
https://github.com/Snektron/vulkan-zig/pull/73#issuecomment-1407523401

Reply:
I don't recall seeing any enums marked as optional.

Pointers are a little trickier as they're not directly marked as optional, and thus vulkan-zig does not mark them as optional. Instead the `ptrCount` field is marked as optional, indicating the pointer is optional as well.

Reply:
Hm yes, i propose to leave this for now. These fields really should be nullable in order to deal with bad drivers

Reply:
@ashpil I think its best to remove the default undefined initialization. After that this pr seems fine to merge to me. Sorey for the delay